docs/source/projects.rst

```diff
@@ -349,103 +349,170 @@ useful if you quickly want to test a project in your existing shell environment.
+
+   .. code-block:: bash
+
+    mlflow run <project_uri> --backend kubernetes --backend-config examples/docker/kubernetes_config.json
+
+   where ``<project_uri>`` is a Git repository URI or a folder.
+
+.. _kubernetes_execution_job_templates:
+
+Job Templates
+~~~~~~~~~~~~~
+
+MLflow executes Projects on Kubernetes by creating `Kubernetes Job resources
+<https://kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion/>`_.
+MLflow creates a Kubernetes Job for an MLflow Project by reading a user-specified
+`Job Spec
+<https://kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion/#writing-a-job-spec>`_.
+When MLflow reads a Job Spec, it formats the following fields:
+
+- ``metadata.name`` Replaced with a string containing the name of the MLflow Project and the time
+  of Project execution
+- ``spec.template.spec.container[0].name`` Replaced with the name of the MLflow Project
+- ``spec.template.spec.container[0].image`` Replaced with the URI of the Docker image created during
+  Project execution. This URI includes the Docker image's digest hash.
+- ``spec.template.spec.container[0].command`` Replaced with the Project entry point command
+  specified when executing the MLflow Project.
+
+The following example shows a simple Kubernetes Job Spec that is compatible with MLflow Project
+execution. Replaced fields are indicated using bracketed text.
+
+.. rubric:: Example Kubernetes Job Spec
 
-- ``metadata.name``;
-
-- ``spec.template.spec.container[0].name``;
-
-- ``spec.template.spec.container[0].image``;
-
-- ``spec.template.spec.container[0].command``;
-
-
-.. code-block:: json
+.. code-block:: yaml
 
   apiVersion: batch/v1
   kind: Job
   metadata:
-    name: ""
+    name: "{replaced with MLflow Project name}"
     namespace: mlflow
   spec:
     ttlSecondsAfterFinished: 100
     backoffLimit: 0
     template:
       spec:
         containers:
-        - name: "name"
-          image: "image"
-          command: ["cmd"]
+        - name: "{replaced with MLflow Project name}"
+          image: "{replaced with URI of Docker image created during Project execution}"
+          command: [{replaced with MLflow Project entry point command}]
         resources:
           limits:
             memory: 512Mi
           requests:
             memory: 256Mi
         restartPolicy: Never
 
-To run your project just use the command:
-
-.. code-block:: bash
-
-  mlflow run <uri> --backend kubernetes --backend-config examples/docker/kubernetes_config.json
-
-Where ``<uri>`` is a Git repository URI or a folder.
-
-To see it in action, you can use the `Docker example <https://github.com/mlflow/mlflow/tree/master/examples/docker>`_ 
-with the ``kubernetes_backend.json`` and ``kubernetes_job_template.yaml`` files.
-
+Note that the ``container.name``, ``container.image``, and ``container.command`` fields are only
+replaced for the *first* container defined in the Job Spec. All subsequent container definitions
```

Comment:
Note that the ``container.name`` -> The ``container.name``

Reply:
Done!